### PR TITLE
Enum structs

### PR DESCRIPTION
**Pull request by Desdaemon**

### CODE REVIEW

**Comment by fzyzcjy** [comment]
Sounds great! Shall I merge now?

**Comment by fzyzcjy** [comment]
By the way, maybe the PR title should be changed?

File: frb_example/pure_dart/dart/pubspec.yaml [hunk 1]
```diff
@@ -11,4 +11,7 @@ dependencies:
     path: ../../../frb_dart
   test: ^1.18.2
   vm_service: ^7.3.0
+  freezed_annotation:
 dev_dependencies:
+  build_runner:
+  freezed:
```
Comment (fzyzcjy) [inline frb_example/pure_dart/dart/pubspec.yaml:17]: What about giving it a version (e.g. `freezed: ^1.1.0`)

File: frb_example/pure_dart/dart/lib/main.dart [hunk 2]
```diff
@@ -263,6 +263,30 @@ void main(List<String> args) async {
...
+    {
+      final res = await Future.wait([
+        KitchenSink.empty(),
+        KitchenSink.boxed(42),
+        KitchenSink.enums(Weekdays.Monday),
+        KitchenSink.nested(KitchenSink.empty()),
+        KitchenSink.buffer(Uint8List.fromList(const [1, 2, 3])),
+        KitchenSink.optional(42),
+        KitchenSink.structlike(foo: Foobar.baz(name: 'John Doe'), bar: 123),
+      ].map((ty) => api.handleComplexEnum(val: ty)));
```
Comment (fzyzcjy) [inline frb_example/pure_dart/dart/lib/main.dart:286]: Maybe we can add some assertions (`expect(...)`)

**Comment by Desdaemon** [comment]
> By the way, maybe the PR title should be changed?

I'm not too sure, I guess it still falls under enum structs? Even though this is more of a refactoring/fixing stuff.

**Comment by fzyzcjy** [comment]
> I'm not too sure, I guess it still falls under enum structs? Even though this is more of a refactoring/fixing stuff.

Sure. By the way, the accidentally merged PR in the morning was *reverted* after it was merged, so this current PR is indeed fully implementation of enum structs.

**Comment by fzyzcjy** [comment]
Shall I merge now?

**Comment by fzyzcjy** [comment]
https://github.com/fzyzcjy/flutter_rust_bridge/releases/tag/v1.10.0 :tada:

**Pull request merged**